substra/sdk/backends/local/backend.py

```diff
@@ -328,9 +339,10 @@ def _add_composite_algo(self, spec, exist_ok, spec_options=None):
     def _add_dataset(self, spec, exist_ok, spec_options):
         self.__check_metadata(spec.metadata)
         permissions = self.__compute_permissions(spec.permissions)
+        key = fs.hash_file(spec.data_opener)
         asset = models.Dataset(
-            key=fs.hash_file(spec.data_opener),
-            pkhash=fs.hash_file(spec.data_opener),
+            key=key,
+            pkhash=key,
```

Comment:
Why do we need `key` and `pkhash` fields?

Reply:
I thought that the remote Dataset returned key and pkhash, actually it has only key, changed it :)

Reply:
But the testDataSampleKeys and trainDataSampleKeys can be None, so in models.py I put a default value of `list()` for these fields instead of making them optional.

Reply:
Thanks, because of your comment I realised that many of the `list_` functions failed, the behaviour with the `pkhash` not defined is actually common to all objects that have a pkhash.

Reply:
If possible, it would like to avoid having a single `pkhash` field in the SDK responses. This field is in my opinion internal to the backend and should not be exposed to the user. One field for the asset unique id is enough :)

But as the backend have inconsistent responses, I'm not sure if it's doable :)

Reply:
Actually, where does it come from to have both `pkhash` and `key` ? I thought that `pkhash` was in the process of being renamed `key` but it does not seem to be the case.
I can always hide it from the users (`to_response` method), but they have it in 'normal' mode and I know that at least Jean's code relies on it rather than `key`

Reply:
> Actually, where does it come from to have both pkhash and key ?

I think it comes from having responses returned by both the backend and the chaincode and multiple refactoring. Pkhash is internal to django (or DRF), it's an abbreviation for "primary key hash".
If I remember well, `pkhash` was used in the old times and a switch to `key` has been made progressively.

Let's keep it that way, later we could a proper refactoring in the backend / sdk to ensure pkhash is not visible from the user point of view.